src/Data/Text.hs

```diff
@@ -1881,6 +1883,43 @@ isInfixOf needle haystack
     | otherwise          = not . L.null . indices needle $ haystack
 {-# INLINE [1] isInfixOf #-}
 
+-- 2021-09-29: NOTE:
+--  * after the implementation - determine & mention the big O
+-- | The 'isSubsequenceOf' function takes the main text and the subsequnce
+-- to find and returns 'True' iff the second argument is a subsequence
+-- of the first.
+--
+-- "Subsequence" used in the meaning of: characters of the second argument
+-- appear in same sequential order in the main data, to say second argument can
+-- be derived by deleting some (any) or no elements from the first.
+--
+-- Examples:
+--
+-- >>> isSubsequenceOf "1234567" "1356"
```

Comment:
Aren't the arguments her backwards? Generally `isInfixOf` etc. are used infix:

```haskell
>>> "ello" `isInfixOf` "Hello, world!"
True
```
but this would be
```haskell
"1234567" `isSubsequenceOf` "1356"
```
which should be False, if you read it in English. It should be `isSubsequenceOf needles haystack` - `"1234567"` _is not_ a subsequence of `"1356"`.